pkg/tlsf/Makefile.dep

```diff
@@ -0,0 +1,7 @@
+ifneq (,$(filter tlsf-malloc,$(USEMODULE)))
+  ifneq (,$(filter newlib,$(USEMODULE)))
+    USEMODULE += tlsf-malloc_newlib
+  else
+    USEMODULE += tlsf-malloc_native
```

Comment:
Maybe here enforce that `BOARD` == `native`.
Would prevent being in this branch with other boards that do not have `newlib`.

Not sure what to do for other boards.